File: Task-03/ELixir/subtask4.exs
defmodule DiamondPattern do
  def run do
    # Read the number of rows for the diamond from input.txt
    {:ok, n} = File.read("input.txt")
    n = String.trim(n) |> String.to_integer()

    middle = div(n + 1, 2)

    # Open output.txt for writing
    {:ok, file} = File.open("output.txt", [:write])

    # Generate the upper part of the diamond
    Enum.each(1..middle, fn i ->
      spaces = middle - i
      stars = 2 * i - 1
      IO.write(file, "#{String.duplicate(" ", spaces)}#{String.duplicate("*", stars)}\n")
    end)

    # Generate the lower part of the diamond
    Enum.each((middle - 1)..1, fn i ->
      spaces = middle - i
      stars = 2 * i - 1
      IO.write(file, "#{String.duplicate(" ", spaces)}#{String.duplicate("*", stars)}\n")
    end)

    # Close the file
    File.close(file)
  end
end

DiamondPattern.run()

File: Task-03/Ruby/subtask4.rb
# Read the value of n from input.txt
input= File.read('input.txt')
n=input.read.to_i
input.close()

# Open the output.txt file for writing
File.open('output.txt', 'w') do |file|
  middle = (n / 2) + 1

  (1..middle).each do |i|
    line = ' ' * (middle - i) + '*' * (2 * i - 1)
    file.puts(line)
  end

  (middle-1).downto(1) do |i|
    line = ' ' * (middle - i) + '*' * (2 * i - 1)
    file.puts(line)
  end
end
